wagtail/admin/templates/wagtailadmin/notifications/task_state_approved.html

```diff
@@ -3,6 +3,26 @@
 {% base_url_setting default="" as base_url %}
 
 {% block content %}
-    <p>{% blocktrans trimmed with title=page.get_admin_display_title task=task.name %}The page "{{ title }}" has been approved in moderation stage "{{ task }}".{% endblocktrans %}</p>
-    <p>{% trans "You can edit the page here:"%} <a href="{{ base_url }}{% url 'wagtailadmin_pages:edit' page.id %}">{{ base_url }}{% url 'wagtailadmin_pages:edit' page.id %}</a></p>
+    {% if page %}
+        <p>{% blocktrans trimmed with title=page.get_admin_display_title task=task.name %}The page "{{ title }}" has been approved in moderation stage "{{ task }}".{% endblocktrans %}</p>
+        <p>{% trans "You can edit the page here:"%} <a href="{{ base_url }}{% url 'wagtailadmin_pages:edit' page.id %}">{{ base_url }}{% url 'wagtailadmin_pages:edit' page.id %}</a></p>
+    {% else %}
+        {% latest_str object as title %}
+        {% admin_edit_url object user as edit_url %}
+        <p>
+            {% blocktrans trimmed with title=title task=task.name model_name=model_name asvar message %}
+                The {{ model_name }} "{{ title }}" has been approved in moderation stage "{{ task }}".
+            {% endblocktrans %}
+            {{ message|capfirst|safe }}
```

Comment:
Does the message need to be marked as safe here? I can't see any html called for.

Reply:
You're right for the HTML files, but for the `.txt` files we need them as otherwise characters like `"` get escaped as `&quot;`, which will show up as-is in the email.